eos/utils/expression-visitors.cc

```diff
@@ -385,20 +386,46 @@ namespace eos::exp
     {
         std::set<std::string> kinematic_set;
 
-        const auto & kinematics_values = e.kinematics_specification.values;
-        const auto & kinematics_aliases = e.kinematics_specification.aliases;
+        const auto & entries = impl::observable_entries;
+        const auto & it = entries.find(e.observable_name);
 
-        // Set or alias kinematic specifications
-        for (const auto & value : kinematics_values)
+        // check if 'e' matches the name of a known observable
+        if (it != entries.end())
         {
-            kinematic_set.insert(value.first);
+            const auto entry = it->second;
+
+            kinematic_set.insert(entry->begin_kinematic_variables(), entry->end_kinematic_variables());
+
+            const auto & kinematics_values = e.kinematics_specification.values;
+            const auto & kinematics_aliases = e.kinematics_specification.aliases;
+
+            // Set or alias kinematic specifications
+            for (const auto & value : kinematics_values)
+            {
+                kinematic_set.insert(value.first);
```

Comment:
Shouldn't we *remove* the kinematic variable from the list since it has been fixed?